src/firefly/energy.py

```diff
@@ -22,12 +59,20 @@ class EnergyDisplay(display.FireflyDisplay):
         "background: rgb(220, 53, 69); color: white; border-color: rgb(220, 53, 69)"
     )
     stylesheet_normal = ""
+    energy_positioner = None
```

Comment:
For line 56 and 57, the max and min energy for Si (111) is from 4 to 40 keV, for ML it is 5~22 depending on the gap? Should we distinguish ML from (111) here?

Reply:
I will re-write this to get the min/max values from the monochromator.

Eventually, we will also need a check when setting the energy to ensure it's within range for the mono/ID/2° mono.

For now, we'll use 5 -- 38keV for 25-ID.

Reply:
I decided to derive the range of valid energies from the limits on the setpoint of the mono energy and ID energy. This way, we can set those values properly in EPICS and everything just kind of works.

Right now the range is 10 eV to 35000 eV, which is wrong, but it's wrong because the limits are not set properly. The current limits are:

- **mono:** 0 ev to 35,000 eV
- **undulator**: 0.01 keV to 100 keV